src/libraries/System.Private.CoreLib/src/System/Collections/Generic/ArraySortHelper.cs

```diff
@@ -34,13 +25,13 @@ internal static int FloorLog2PlusOne(int n)
             while (n >= 1)
             {
                 result++;
-                n /= 2;
```

Comment:
Was the JIT not getting this optimization? If not, it seems like something we should have a bug on.

Reply:
Also, would this be better handled by `BitOperations.Log2` which uses hardware intrinsics when available?

Reply:
> Was the JIT not getting this optimization?

No.  But `n` here is `int` rather than `uint`... it's not supposed to be able to do that transformation, is it?

Reply:
> would this be better handled by BitOperations.Log2 which uses hardware intrinsics when available?

Maybe? :smile:

Reply:
> But n here is int rather than uint (even though it's never going to be negative)... it's not supposed to be able to do that transformation, is it?

I would think it is safe if the JIT can assert that it is always positive. I don't know if that's something it easily supports today.

Reply:
> Maybe?

I believe we are already returning the `floor` for `BitOperations.Log2`, in which case this should just be `return BitOperations.Log2(n) + 1`

Reply:
It ends up having the same result for all inputs > 0 (but different for 0).  Given that all inputs in our use of this are positive, I'll just switch the call sites to using BitOperations.Log2 and delete this.